pageserver/src/page_service.rs

```diff
@@ -405,17 +394,22 @@ impl PageServerHandler {
             .get_timeline(timeline_id, true)
             .map_err(|e| anyhow::anyhow!(e))?;
 
+        // Avoid starting new requests if the timeline has already started shutting down,
+        // and block timeline shutdown until this request is complete, or drops out due
+        // to cancellation.
+        let _timeline_guard = timeline.gate.enter().map_err(|_| QueryError::Shutdown);
```

Comment:
Missing questionmark?

Reply:
c2a7c67f22516be713bcb6bb291e73c6769f6774

Weird, I would have expected Result's `must_use` to help with this